### PR TITLE
send-info-after-death

### LINKED ISSUE #2896: Coords in Death Message
### Feature request

**Feature description**
I would like an option to include death coords in death messages. Really, just some way to let players know where they died.

**How the feature is useful**
As the owner of an smp, players often ask me to tell them their death coords. It would make it easier on me if players could find the coords on their own.

### PR DESCRIPTION
**Pull request by kaanakduman**
A possibly fixed version of https://github.com/EssentialsX/Essentials/pull/2912. I'm not very skilled in coding, but here's my best shot on making the suggested changes. I couldn't have done this without seeing the PR from @darbyjack . I'm not sure if this is the standard way of modifying a PR, sorry if it's not.

Closes #2896, closes #3207

### CODE REVIEW

**Review by pop4959 — APPROVED** [review]

**Comment by pop4959** [comment]
I don't know if there is a convenient way to modify other people's pull requests without being a project maintainer. This is probably fine, but I think @darbyjack should at least have a chance to take a look at this since he has the original pull request still open.

As far as I can tell, it seems all of the suggested fixes were applied appropriately though. Looks good.

**Comment by kaanakduman** [comment]
Thanks pop! Could you help me understand something? Why did the check fail for me when it passed for the other PR? Also, how come you approved the changes even though the check failed?

**Comment by pop4959** [comment]
@kakd The build failed because Spigot added new methods to interfaces which Essentials is using in some of its implementation / testing framework. If you look at the details of the build you will see this:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:compile (default-compile) on project EssentialsX: Compilation failure
[ERROR] /home/runner/work/Essentials/Essentials/Essentials/src/com/earth2me/essentials/OfflinePlayer.java:[41,8] com.earth2me.essentials.OfflinePlayer is not abstract and does not override abstract method swingOffHand() in org.bukkit.entity.LivingEntity
```
Someone will just have to run a re-build of this PR once those methods are added. Nothing needed on your part.

**Comment by pop4959** [comment]
@kakd Build fails for a new reason now. Apparently it isn't able to check out from your repo anymore. I'm not sure why that would be the case but you should probably check on that.

```
reference is not a tree: b87981a785670f5daf6f5b8308dbf998206d0996
Git checkout failed with exit code: 128
```

File: Essentials/src/com/earth2me/essentials/EssentialsEntityListener.java [hunk 1]
```diff
@@ -141,13 +142,17 @@ public void onEntityCombustByEntity(final EntityCombustByEntityEvent event) {
...
+        final Location loc = user.getLocation();
         if (user.isAuthorized("essentials.back.ondeath") && !ess.getSettings().isCommandDisabled("back")) {
             user.setLastLocation();
             user.sendMessage(tl("backAfterDeath"));
         }
         if (!ess.getSettings().areDeathMessagesEnabled()) {
             event.setDeathMessage("");
         }
+        if (ess.getSettings().infoAfterDeath()) {
+            user.sendMessage(tl("infoAfterDeath", loc.getWorld(), loc.getBlockX(), loc.getBlockY(), loc.getBlockZ()));
```
Comment (Ichbinjoe) [inline Essentials/src/com/earth2me/essentials/EssentialsEntityListener.java:154]: This should be log.getWorld().getName or something like that

**Review by Ichbinjoe — CHANGES_REQUESTED** [review]

**Comment by SrBedrock** [comment]
@kakd

**Comment by darbyjack** [comment]
Closing this one as you've recreated a new one.

**Pull request closed without merging**